### PR TITLE
Issue/simd (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by artv3**
Collecting examples of failed vectorization.

### CODE REVIEW

**Comment by artv3** [comment]
@ajkunen I got the ball rolling with a couple of examples. Are you looking at the assembly code or the Intel vectorization report?

**Comment by davidbeckingsale** [comment]
It's a good idea to collect things like this. Is there a programmatic way to collect a pass/fail output from each example? If so, we can add these to the tests.

**Comment by rhornung67** [comment]
@davidbeckingsale @ajkunen @artv3 does it make sense to set up a different test suite (and associated 'make test' target) for tests like this that reveal issues?

File: simd-issues/example3.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  long int *A = new long int[arrLen]; 
+  long int *B = new long int[arrLen];
+  long int *C = new long int[arrLen];
+
+  for(auto i=0; i<arrLen; ++i){
+    A[i] = i; B[i] = -i; C[i] = 0;
+  }
+  
+  RAJA::View<long int, RAJA::Layout<3> > Aview(A,n1,n2,n3);
```
Comment (davidbeckingsale) [inline simd-issues/example3.cpp]: Should use `Index_type`

**Comment by artv3** [comment]
@ajkunen , I was thinking about these examples and I wondering about your experiment setup. Could we meet this week? I created a main.cpp file where I wanted to compare performance on seq, loop, and simd for various examples.

**Comment by artv3** [comment]
@jeffhammond we found a combination of policies that breaks correctness with the Intel compiler (Intel 17). Could you take a look at it? The example is called main.cpp it is found in the folder simd-issues. If there is anything I can do to assist or any questions please let me know.

**Comment by jeffhammond** [comment]
@artv3 How did you know that I was working with RAJA at this exact moment? 😄   Yes, I'll take a look.  Have you tried Intel 18?

**Comment by artv3** [comment]
Thanks! I did, and I believe the issue is still there.

**Comment by jeffhammond** [comment]
You probably know this, but the problem is the `_Pragma` below, which means it should be pretty easy to create a MCVE that isn't 85KLOC.  I haven't used [C-Reduce](https://embed.cs.utah.edu/creduce/) before.  Have you?
```c++
template <typename Iterable, typename Func>
inline __attribute__((always_inline)) void forall_impl(const simd_exec &,
                             Iterable &&iter,
                             Func &&loop_body)
{
  auto begin = std::begin(iter);
  auto end = std::end(iter);
  auto distance = std::distance(begin, end);
  _Pragma("omp simd")
  for (decltype(distance) i = 0; i < distance; ++i) {
    loop_body(*(begin + i));
  }
}
```

**Comment by artv3** [comment]
@jeffhammond , Can you point me to the code? I am not sure where the code snippet is.

**Comment by jeffhammond** [comment]
```sh
$ git grep "void forall_impl(const simd_exec"
include/RAJA/policy/simd/forall.hpp:RAJA_INLINE void forall_impl(const simd_exec &,
```

**Comment by jeffhammond** [comment]
I tried C-reduce but did it wrong.  I think I can fix it but don't have time right now.

**Comment by artv3** [comment]
@jeffhammond , Why is the following pragma an issue "_Pragma("omp simd") " as referenced above ?

**Comment by jeffhammond** [comment]
I don't think the pragma is an issue, but rather the combination of the pragma and the code to which it applies.  Sadly, I have not had any time to work on this issue.

**Comment by artv3** [comment]
I added a second example that breaks correctness (and the fix for it), it does require using the reduction clause to generate correct results. I suspect that this is the culprit that is generating incorrect results for @Ram3sh 's code.

**Pull request closed without merging**